app/main.py

```diff
@@ -174,18 +143,39 @@ def format_datum(datum):
             count=int(count)
         ) for date, lat, long, count in datum]
 
-    with conn.cursor() as cursor:
-        cursor.execute(SQL)
-        data = cursor.fetchall()
-        return PlantGrowthDataResponse(
-            plant_growth_data=[PlantIdData(species=s, plant_growth_datum=format_datum(d)) for s, d in data]
-        )
+    species_data = get_species_data(conn, center_lat, center_lon, scan_range, day_range)
+    return PlantGrowthDataResponse(
+        plant_growth_data=[PlantIdData(species=s, plant_growth_datum=format_datum(d)) for s, d in species_data]
+    )
+
+
+# Get the daily growth of each species
+@app.get("/estimate_species_gradients")
+def estimate_species_gradients():
+    return get_species_daily_growth(conn)
+
+
+# Get the pattern of species growth
+@app.get("/estimate_growth_pattern")
+def estimate_growth_pattern():
+    THRESHOLD = 0.05  # Magic number
+    species_gradients = get_species_daily_growth(conn)
+    species_patterns = {}
+
+    for species, gradient in species_gradients.items():
+        if abs(gradient) < THRESHOLD:
+            species_patterns[species] = "CONSTANT"
```

Comment:
Should this one be decaying and line 171 be constant?

Reply:
ah right, my mistake, as you said I failed to notice this one had abs and line 168 doesn't